### PR TITLE
Updates to the intent-based ansible tutorial

### PR DESCRIPTION
**Pull request by wdesmedt**

### CODE REVIEW

**Comment by wdesmedt** [comment]
@hellt The CI run fails on files that were not changed in the commits of this PR, e.g. tutorials/programmability/json-rpc/basics/index.html. I guess I can ignore these?

**Review by hellt — DISMISSED** [review]
@wdesmedt yes, you can ignore the CI errors, I will fix them in a separate PR.

Merge this PR when you are ready

**Comment by wdesmedt** [comment]
@hellt Fred is doing extra review. I'll mark PR ready-for-review when done.

File: docs/tutorials/programmability/ansible/intent-based-management/project-structure.md [hunk 1]
```diff
@@ -31,151 +23,73 @@ The `ansible-inventory.yml` defines four groups:
...
+1. the `hosts` variable at play-level defines the hosts that are part of the fabric. In this case, all hosts in the `leaf` and `spine` groups. Group definition and membership is defined in the inventory file.
+2. the `roles` variable defines the roles that are applied to the hosts defined in the `hosts` section. The roles are applied in the order they are defined in the playbook. The roles are grouped in 4 sections: `INIT`, `INFRA`, `SERVICES` and `CONFIG-PUSH`.
+    - **INIT**: This section initializes some extra global variables or _Ansible facts_ that are used by other roles. These facts include:
+        - the current 'running config' of the device
+        - the SR Linux software version
+        - the LLDP neighborship states
+    - **INFRA**: This section has 2 roles:
+        - `fabric`: this role fully configures the fabric based on a fabric intent defined in the intent directory. If there is no fabric intent file present, this role will have no effect (skipped)
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply docs/tutorials/programmability/ansible/intent-based-management/project-structure.md:89]: fixed

File: docs/tutorials/programmability/ansible/intent-based-management/config.md [hunk 2]
```diff
@@ -66,388 +64,670 @@ If you have installed the [fcli](https://github.com/srl-labs/nornir-srl) tool, y
...
+    location: 'spine'
+    # List of neighbor IPs for Route Reflectors in case of 'external'
+    # neighbor_list:
+    #   - 1.1.1.1
+    #   - 2.2.2.2
+
+  # Spine specific configuration, defines which ports can be used for ISLs
+  spine:
+    clab-4l2s-s[1-2]:
+      isl-ports: ethernet-1/[1-128]
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply docs/tutorials/programmability/ansible/intent-based-management/config.md:106]: @freddebacker I'll add a comment that this is for H2. It doesn't affect the result as the `fabric` role selects ISL ports on spines from e-1/1 onwards.

File: docs/tutorials/programmability/ansible/intent-based-management/config.md [hunk 3]
```diff
@@ -66,388 +64,670 @@ If you have installed the [fcli](https://github.com/srl-labs/nornir-srl) tool, y
...
+!!!note
+    Since we're using BGP in the underlay to provide connectivity for IBGP/EVPN in the overlay, not all BGP sessions will establish immediately. First the underlay EBGP sessions must be established and exchange loopback address information before the IBGP sessions can establish.
+
+!!!note
+    Notice that the IBGP sessions exchange no routes. This is expected as we didn't configure overlay services or Ethernet Segments yet that would trigger announcement of EVPN routes.
+
+Alternatively, we could opt to use _BGP unnumbered_ inter-switch links (dynamic BGP-peers using IPv6 LLA addresses). It suffices to change the `.fabric.underlay_routing.bgp.bgp-unnumbered` from `false` to `true`, remove the `.fabric.p2p` parameter as inter-switch link addresses are no longer needed, and run the playbook again. Since intents are declarative, there is no need to worry about the previous configuration. Run the playbook again as before and check the fabric status.
+
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply docs/tutorials/programmability/ansible/intent-based-management/config.md:268]: done.

File: docs/tutorials/programmability/ansible/intent-based-management/config.md [hunk 4]
```diff
@@ -66,388 +64,670 @@ If you have installed the [fcli](https://github.com/srl-labs/nornir-srl) tool, y
...
-clab-4l2s-l3               : ok=11   changed=0    unreachable=0    failed=0    skipped=12   rescued=0    ignored=0   
-clab-4l2s-l4               : ok=11   changed=0    unreachable=0    failed=0    skipped=12   rescued=0    ignored=0   
-clab-4l2s-s1               : ok=11   changed=0    unreachable=0    failed=0    skipped=12   rescued=0    ignored=0   
-clab-4l2s-s2               : ok=11   changed=0    unreachable=0    failed=0    skipped=12   rescued=0    ignored=0   
+Copy over above 2 files from `intent_examples/services` into `intent/` and run the playbook:
+
+```
+cp intent_examples/services/mh_access-1.yml intent_examples/services/host_infra_itf_tags.yml intent
+ansible-playbook -i inv -e intent_dir=${PWD}/intent --diff playbooks/cf_fabric.yml
 ```
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply docs/tutorials/programmability/ansible/intent-based-management/config.md:346]: this is expected as we need to enable the interfaces for the lags of the `mh_access` intent.

File: docs/tutorials/programmability/ansible/intent-based-management/config.md [hunk 5]
```diff
@@ -66,388 +64,670 @@ If you have installed the [fcli](https://github.com/srl-labs/nornir-srl) tool, y
...
+  type: mac-vrf
+  description: subnet-2
+  proxy_arp: true
+  interface_tags:
+    - _mh-4
+    - _mh-5
+  export_rt: 100:102
+  import_rt: 100:102
+  vlan: 200
 ```
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply docs/tutorials/programmability/ansible/intent-based-management/config.md:646]: done.

File: docs/tutorials/programmability/ansible/intent-based-management/config.md [hunk 6]
```diff
@@ -66,388 +64,670 @@ If you have installed the [fcli](https://github.com/srl-labs/nornir-srl) tool, y
...
+    PING 10.0.2.3 (10.0.2.3) 56(84) bytes of data.
+    64 bytes from 10.0.2.3: icmp_seq=1 ttl=63 time=0.347 ms
+    64 bytes from 10.0.2.3: icmp_seq=2 ttl=63 time=0.200 ms
+    64 bytes from 10.0.2.3: icmp_seq=3 ttl=63 time=0.205 ms
+
+    --- 10.0.2.3 ping statistics ---
+    3 packets transmitted, 3 received, 0% packet loss, time 2039ms
+    rtt min/avg/max/mdev = 0.200/0.250/0.347/0.068 ms
+    ```
+=== "ARP table`"
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply docs/tutorials/programmability/ansible/intent-based-management/config.md:698]: fixed.

**Comment by wdesmedt** [comment]
@hellt this is ready to be merged finally.

**Pull request merged**